models/models.go

```diff
@@ -1,17 +1,62 @@
 package models
 
 import (
+	"fmt"
+	"net/url"
+
 	"github.com/rivo/tview"
 )
 
 type Connection struct {
-	Name     string
-	Provider string
-	DBName   string
-	URL      string
+	Name string
+
+	// either use this directly
+	URL string
+
+	// or parse manually
+	Provider  string
+	Username  string
+	Password  string
+	Hostname  string
+	Port      string
+	DBName    string
+	URLParams string
+
 	Commands []*Command
 }
 
+// ParseURL will manually parse url if url empty
+//
+// for handling username & password with special characters
+//
+// only sqlserver for now
+//
+// need to refactor if wanted to reuse driver list in drivers/constants.go
+func (c *Connection) ParseURL() {
+	if c.URL != "" {
+		return
+	}
+
+	// only sqlserver for now
+	if c.Provider != "sqlserver" {
```

Comment:
You defined a constant, you can use it

Reply:
![image](https://github.com/user-attachments/assets/adf929e1-7e82-42d1-94da-16c7dd8351d8)

i wanted but i cant

Reply:
IT happens, sometimes can be moved in other packages, like a dialect at root level
But it leads to a lot of code change, not fun when dealing with a lib, but here it's a CLI

Can't this method moved in another package ?

Reply:
i think i can move it into `helpers/config.go`